prog/weaveutil/db_flag.go

```diff
@@ -0,0 +1,54 @@
+// Get/set arbitrary flags in the Weave Net persistence DB
+package main
+
+import (
+	"fmt"
+
+	"github.com/weaveworks/weave/db"
+)
+
+const dbFlagPrefix = "flag:"
+
+func getDBFlag(args []string) error {
+	if len(args) != 2 {
+		cmdUsage("check-db-flag", "<db-prefix> <flag-name>")
+	}
+	dbPrefix := args[0]
+	flagName := args[1]
+
+	d, err := db.NewBoltDBReadOnly(dbPrefix)
+	if err != nil {
+		return err
+	}
+	defer d.Close()
+	var value string
+	nameFound, err := d.Load(dbFlagPrefix+flagName, &value)
+	if err != nil {
+		return err
+	}
+	if !nameFound {
+		return fmt.Errorf("")
```

Comment:
Returning an empty error to signal we haven't found something seems unintuitive. Would be much clearer to have `getDBFlag` return the value instead of printing it out.

Reply:
Printing the value is the pattern of all functions in `weaveutil`; I doubt it's worth breaking that pattern for this small point.

The empty error is along the lines of `if value, ok := foo[bar]; ok`